frame/nomination-pools/src/lib.rs

```diff
@@ -2117,6 +2138,85 @@ pub mod pallet {
 			ensure!(bonded_pool.can_nominate(&who), Error::<T>::NotNominator);
 			T::Staking::chill(&bonded_pool.bonded_account())
 		}
+
+		/// Bond pending rewards of `member_account` into the pool they already belong.
+		///
+		/// Note: `member_account` must pass `RewardClaim::Permissionless` to `set_reward_claim`,
+		/// making this call permissionless.
+		#[pallet::call_index(14)]
+		#[pallet::weight(
+			T::WeightInfo::bond_extra_transfer()
+			.max(T::WeightInfo::bond_extra_reward())
+		)]
+		pub fn bond_extra_pending_rewards_other(
```

Comment:
I may have asked this before, but can't this be one transaction with the existing `bond_extra`? and even if we want to keep them separate, shan't they share most of the internal code?

Reply:
changing the transactional format of `bond_extra` would require a runtime upgrade.

it does share most of the internal code of  `bond_extra`, it excludes transferring pending rewards to the member from the `do_reward_payout` helper function and instead transfers the pending reward directly to the bounding account from `try_bound_fund` and increment the points accordingly.

Reply:
But I believe we can further improve auto compounding in this PR by restricting `unbound` and `claim_payout` till permission is reverted back to the member.

Reply:
@kianenigma Yes an `option` can be added to the transaction format of `bond_extra` instances where `Origin` is the focus will be changed to `Options<AccountId>`